game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/pbf/ForumPosterEditor.java

```diff
@@ -41,13 +41,15 @@
   private final JTextField usernameField = new JTextField();
   private final JLabel passwordLabel = new JLabel("Forum Password");
   private final JPasswordField passwordField = new JPasswordField();
+  private final JLabel otpLabel = new JLabel("2FA OTP-Code (optional)");
```

Comment:
`otp` is not that well known of an acronym. What is it spelled out?

Can we add a help text or box next to this as well to explain what this is for a user? We *do* want people to use this after all. If we look at the 1.9 PBF menu, we can see that the game explained pretty much all the options quite well. This meant any user could learn how to use PBF, even if it was kinda clunky.

2FA OTP-code is a bit cryptic to even me. I think we'll need to help users a bit more, and/or explain when/why they'd want to use this. I'm a bit curious myself when we would want to use OTP code and not just use passwords.

Reply:
OTP is short for "One-Time-Password".

This Field is only required for people that have 2FA enabled in the Forum. Up until recently the "write-api" for NodeBB didn't check if 2FA was enabled and just ignored it. Because @panther2 pointed this out I created an issue which lead to a fix. Currently people with 2FA enabled can't use PbF at all and this extra field should fix this

Reply:
Also perhaps worth mentioning: The OTP isn't a replacement for the password, it is a necessary addition

Reply:
Cool, that is good context. Do you have a plan for how to convey that information to users? Specifically, what the field is, when it is needed, and if then not already clear - perhaps where to get their OTP when it is needed.

Reply:
Not really, but I think this is a problem that isn't limited to TripleA. How can we explain or even advertise 2FA for users?
For most people this just adds another inconvenient step to the login process, and I'd assume most people already using 2FA know what to look out for, they don't need any extra explanation, they just want a field to enter their newly generated code.

If you have any good explanation, please let me know. I'd also like to know how to explain such an "abstract" concept in such a limited screen space 🤔

Reply:
To clarify: Obviously we could implement a wall of text, but I bet most people aren't going to read it if it's more than a couple of lines

Reply:
A help button that says this field is used if "2FA is enabled. That this value can be found in forum settings. Doesn't have to be anything to complicated. Leaving it as 'OTP' though is going to be cryptic.